core/common/boot/src/main/java/org/eclipse/dataspaceconnector/boot/system/DefaultServiceExtensionContext.java

```diff
@@ -39,6 +39,7 @@ public class DefaultServiceExtensionContext implements ServiceExtensionContext {
 
     private final Map<Class<?>, Object> services = new HashMap<>();
     private final List<ConfigurationExtension> configurationExtensions;
+    private boolean isFrozen = false;
```

Comment:
This is really my inner monk talking,  but since we're referring to this as "read-only mode", personally i think the variable should be called `isReadOnly`.